aws/table_aws_waf_rule_group.go

```diff
@@ -0,0 +1,286 @@
+				Transform:   transform.FromField("Name", "RuleGroup.Name"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listTagsForAwsWafRuleGroup,
+				Transform:   transform.FromField("TagInfoForResource.TagList").Transform(classicRuleGroupTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getWafRuleGroupArn,
+				Transform:   transform.FromValue().Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsWafRuleGroups(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := WAFService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_waf_rule_group.listAwsWafRuleGroups", "service_creation_error", err)
+		return nil, err
+	}
+
+	// List all rule groups
+	pagesLeft := true
+	params := &waf.ListRuleGroupsInput{
+		Limit: aws.Int64(100),
+	}
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < *params.Limit {
+			if *limit < 1 {
+				params.Limit = aws.Int64(1)
+			} else {
+				params.Limit = limit
+			}
+		}
+	}
+
+	for pagesLeft {
+		response, err := svc.ListRuleGroups(params)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_waf_rule_group.listAwsWafRuleGroups", "api_error", err)
+			return nil, err
+		}
+
+		for _, ruleGroups := range response.RuleGroups {
+			d.StreamListItem(ctx, ruleGroups)
+
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		if response.NextMarker != nil {
+			pagesLeft = true
+			params.NextMarker = response.NextMarker
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsWafRuleGroup(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var id string
+	if h.Item != nil {
+		data := classicRuleGroupData(h.Item, ctx, d, h)
+		id = data["rule_group_id"]
+	} else {
+		id = d.KeyColumnQuals["rule_group_id"].GetStringValue()
+	}
+
+	// Create Session
+	svc, err := WAFService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_waf_rule_group.getAwsWafRuleGroup", "service_creation_error", err)
+		return nil, err
+	}
+
+	params := &waf.GetRuleGroupInput{
+		RuleGroupId: aws.String(id),
+	}
+
+	op, err := svc.GetRuleGroup(params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_waf_rule_group.getAwsWafRuleGroup", "api_error", err)
+		return nil, err
+	}
+
+	return op.RuleGroup, nil
+}
+
+func getAwsWafRuleGroupActivatedRules(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var id string
```

Comment:
Hydrate functions other than getConfig one will always get the required parameters from h.Item. They will not get the required parameters from KeyColumnQuals